### ISSUE TITLE
Is deduplication necessary (especially for highly repetitive genomes)

### ISSUE THREAD

**Issue by varunorama** [FelixKrueger/Bismark#400]
Hello Felix, 

I have a question that is less technical, and maybe more methodology, but I am working with a highly repetitive genome and analyzing WGBS datasets. I am using the Bismark pipeline and mapped the reads using bowtie2 within the Bismark pipeline, and used the "Dirty Harry" approach (PE alignment and then SE alignments for forward and reverse reads, separately). 

I know that deduplication is typically done on WGBS datasets in order to remove PCR-based duplication. I ran this deduplication step (deduplicate_bismark) and then extracted methylation statistics based on the deduplicated data and the non-deduplicated data to see how the data differed. I found that deduplication keeps 55% of the data, suggesting that the ~45% of the data removed belong to PCR duplicated artefacts. Furthermore, the overall coverage of CpG sites is greatly reduced; from an average coverage of 4x (for the non-deduplicated data) to about 1x (for deduplicated data).

Given the large reduction in data and coverage, and that I am working with an organism with a highly repetitive genome, I am wondering if deduplication should indeed be implemented in this case; is deduplication still needed in this case, or not? Or does a highly repetitive genome composition have no effect on deduplication filtering? 

If there are any QC steps needed to make a more informed decision, I would like to hear them!

Thank you!

***ADDENDUM***

I looked at my FastQC reports and it shows that for my PE reads, I should have about 74% remaining after deduplication, and no major duplication levels detected. Based on this, I am leaning towards forgoing deduplication. Is this is a correct assessment? 

![image](https://user-images.githubusercontent.com/20046656/101675461-d4027b00-3a27-11eb-86ff-29ee5595e4f8.png)

**Comment by FelixKrueger** [comment]
The duplication plot in FastQC works simply on the sequence level of all sequences in the file, and more specifically on the first 50bp in case your reads were longer than that. It stores the first 200,000 different sequences, and follows these sequences to the end of the file. Exact matches of 50bp are counted as duplicates.

The deduplication of `deduplicate_bismark` uses the mapping position instead, so chromosome, start (as well as end for PE files) and orientation of the alignment. In contrast to FastQC, 100% would refer to 100% of all aligned sequences, so if your unique alignment rate was only say 50%, the value you would get here could be substantially different to the one from FastQC.

Just generally, I don't think that the repetitiveness of the genome should have a substantial impact on the duplication level. If sequences were perfect matches to 2 (or more) locations in the genome, such sequences would be scored as ambiguous alignments, and discarded. So, if you find alignments to any position in the genome, they must have been better than to any other region, if only slightly. My gut feeling would be to perform de-duplication in your case.

I appreciate that deduplication will remove a substantial amount of your data, and one would instinctively like to hold on to as many methylation calls as possible. On the other hand, more methylation calls mean higher statistical power to detect changes. If, however, the increased number of calls is simply due to PCR amplification, the statistical power would be artificially inflated - something that you would like to avoid.

In an ideal scenario, the average methylation values should not change substantially between the full and deduplicated files. If they do, this might be because certain repetitive elements with a very distinct methylation profile might be affected primarily (again, these alignments might belong to a certain class of repeats, but the alignments themselves cannot be matching perfectly to several locations). Another option could be that sequences with a specific methylation profile were more duplicated than others, e.g. due to a polymerase preference or the like. I hope it's not that, as this would mean that you've got biases in your library preparation somewhere that depend on the actual methylation level... 

Just to mention something (just ignore this comment if you have already done this): if you did the deduplication using the 'Dirty Harry' method, you need to perform two separate deduplication steps, one for the PE BAM file, and one for the single-end files (they can be merged).

I hope this is makes some sense (it's getting quite late over here...), if not, please let me know.

Cheers, Felix

**Comment by varunorama** [comment]
Felix, as usual thank you for such a thoughtful and helpful response. Yes this makes sense, and gives me more clarity (and support) to continue with the deduplication data. 

Also yes, I made sure to perform deduplication for the PE BAM file and the SE files (I ran the deduplication on the forward and reverse separately, and then merged them afterwards - hopefully that's ok). 

I think given the complexity of the organism's genome, this might be the best we can get per sample. 

Thank you again!
Varun
